### ISSUE TITLE
FEATURE REQUEST Fans start automatically when open a program or game.

### ISSUE THREAD

**Issue by AndreaBaccolini** [Rem0o/FanControl.Releases#2478]
Hi, I would like the GPU fans (or fan of my choice) to run by themselves when I start a game (or program) without having to run them first each time. Thanks

**Comment by Rem0o** [comment]
Is there a reason why you wouldn't create a curve based on temperature instead? That's what the program is meant for.

**Comment by AndreaBaccolini** [comment]
> Is there a reason why you wouldn't create a curve based on temperature instead? That's what the program is meant for.

You didn't understand. The fans start when I open a program and follow the curves I set. So a certain program activates a certain control of one or more fans.

**Comment by Rem0o** [comment]
I get your feature request, it is not the first time I get it. My question is to you, is why.   If your program generates heat, your fan curves will kick in. If your program doesn't generate heat, fan curves won't kick in.  

So _*why*_ do you need to have an other feature that forces the fan on or off based on a process starting ?

**Comment by Rem0o** [comment]
You will find me annoying, but why don't you just leave it on? That's the whole point. 

For eg:
![image](https://github.com/Rem0o/FanControl.Releases/assets/14118956/80934e5b-b061-461e-abfc-186d2a87ff35)

This is mine.  Under load (over 50)  I get the the fan spinning, otherwise it sits at 0 RPM. 

> Because I have noticed at least in my GPU that the fan turns off and on many times if I use these apps with native control. 

Which GPU you got? We can probably handle that properly.  Also show what settings you got in FanControl, ideally screenshots.

**Comment by AndreaBaccolini** [comment]
I had tried setting with yours but it didn't work for me.
I mainly play Guild wars 2 and if I set like yours I would have the fans going off and on all the time so I have to set them to go to idle all the time and then increase with the temperature.

**Comment by Rem0o** [comment]
Have you tried a "trigger" instead of a graph?  Set the idle temp low enough so that while in game it never reaches it, and you'll have the desired behavior.

**Comment by AndreaBaccolini** [comment]
Yes. But the problem remains. Unfortunately in old games I think it's easy for that to happen. That's why it would be useful in these cases that the fan control could be operated on a game-by-game basis. So you could also make ad hoc curves by game/program.

**Comment by t3t3** [comment]
I'd like to add, that there are situations, when a fan in a system should target a device, that doesn't provide any kind of temperature readings (and does also not notably raise common temperatures, when in use); I do, for example, use a bunch of DSP cards, that I'd like to provide some extra airflow once they're get going.

My workaround is to use the File "sensor" along with a scheduled task, which traces application start/stop events, which then writes a particular value into that file (using a PowerShell script)... together with some "sensor math" it allows me to accomplish that task fully automated.

Another reason might simply be, that you'd just like to have more aggressive cooling under certain situations, and (of course) want to automate that...

**Comment by vertigo220** [comment]
I'm not really understanding the need for this, either, except for t3t3's point, but perhaps profiles with the ability to switch them via the tray icon and possible hotkeys would be a solution.

**Comment by Rem0o** [comment]
Just curious.   

Say you programmed ProcessA to load ConfigurationA, and an other process, ProcessB, to load CongfigurationB.

You launch processA, then launch processB,   which configuration is loaded?

**Comment by t3t3** [comment]
... the process, which hosts the currently active window (there can only be one). but that is of course also not the definitive answer.

guess the idea is not to have dozens of profiles set up, but one or two for particular special cases where the question does not even arise (like when steam is running, for example – which would probably not cause such conflicts).

ps: i can get along with my solution posted above, but after all it's about automating things, right?

**Comment by Rem0o** [comment]
> another way would be to allow profile switching via a cli interface.

That's already possible.

See https://getfancontrol.com/docs/

**Comment by t3t3** [comment]
> > another way would be to allow profile switching via a cli interface.
> 
> That's already possible.

ah; been using fan control for a long time already, and had apparently missed the introduction of that feature. thanks!
